### PR TITLE
UI Bugfix: Make disabled QPushButton more distinct

### LINKED ISSUE #5260: Default styling: Disabled PushButtons can't be distinguished from Labels
When using magicgui `PushButton` widgets in napari, the default styling of a disabled button is to not show the box around the button text. This essentially makes the widget look like a text Label widget.

I think this styling is unfortunate:

* Depending on the context, the user may not realize that there is a button that may become activated through some means. 
* The user may mistake the button text as an explanatory label for a neighbouring widget, not as an action.
* It doesn't look nice.

Here is an example where a disabled button looks a bit odd (the button only becomes visible once enough reference points have been selected)
image

Suggestion:
Use a button styling that retains the button look but visually communicates the enabled/disabled status through other means such as greyed out text, or changed bg color. This is standard in many other UI frameworks. I realize that "graying out" the text in the napari context is not that trivial as the default styling already uses various shades of gray.

### PR DESCRIPTION
**Pull request by psobolewskiPhD**
# Description
As noted by @VolkerH disabled PushButtons appear as just text on the background with no hint that it's a button, like a label maybe.
Here's how it looks with `main`
image

In this PR I've put a border on them, dimmed the text, and made the button background more distinct, so there is still a clear button there. 
With this PR:
image

## Type of change

- [x] Bug-fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# References
fixes #5260

# How has this been tested?

- [ ] example: the test suite for my feature covers cases x, y, and z
- [ ] example: all tests pass with my change
- [ ] example: I check if my changes works with both PySide and PyQt backends
      as there are small differences between the two Qt bindings.  

## Final checklist:
- [x] My PR is the minimum possible work for the desired functionality
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If I included new strings, I have used `trans.` to make them localizable.
      For more information see our [translations guide](https://napari.org/developers/translations.html).

### CODE REVIEW

**Comment by Czaki** [comment]
Could you add a screenshot from the light theme?

**Comment by psobolewskiPhD** [comment]
> Could you add a screenshot from the light theme?

Light theme on main:
image
This PR:
image

File: napari/_qt/qt_resources/styles/00_base.qss [hunk 1]
```diff
@@ -83,7 +83,10 @@ QPushButton:checked {
 }
 
 QPushButton:disabled {
-  background-color: {{ background }};
+  background-color: {{ opacity(foreground, 75) }};
+  border: 1px solid;
+  border-color: {{ foreground }};
+  color: {{ opacity(text, 90) }};
```
Comment (Czaki) [inline napari/_qt/qt_resources/styles/00_base.qss:89]:
```suggestion
  background-color: {{ opacity(foreground, 120) }};
  border: 1px solid;
  border-color: {{ foreground }};
  color: {{ opacity(text, 160) }};
```

Could you check these values, and how they look to you?

**Comment by psobolewskiPhD** [comment]
@Czaki For me in dark mode the disabled button is less clearly grayed out:
image
In light mode it's OK?
image

I mean if you have both buttons it's clear, but I think it's regarding the case where there is a  button that's disabled and the user doesn't know it—there might not be a 2nd button that is enabled.

**Comment by Czaki** [comment]
Maybe you are right. Let's wait for the opinions of people who are better at design.

**Comment by VolkerH** [comment]
Thanks for picking this up @psobolewskiPhD !

In terms of styling. I think in a group of buttons where some are enabled and some  are disabled it will always be pretty clear what level of grey is to indicate disabled (by sheer contrast). In the absence of any activated buttons, it is more difficult to tell. (While typing I just noticed that @psobolewskiPhD left a similar remark, so +1), so I would opt for the initial variant (or something even fainter).

File: napari/_qt/qt_resources/styles/00_base.qss (hunk 1 above)
Comment (psobolewskiPhD) [reply napari/_qt/qt_resources/styles/00_base.qss:89]: Bah, I posted in the wrong spot.
For me on Dark Mode the disabled is less clearly disabled, if there is just one button that's disabled and no reference enabled button. So the user could possibly not realize it. So I prefer the original, but we can play with the values some more.

**Comment by isabela-pf** [comment]
Thanks for the tag and for this lovely PR! I’m overjoyed to see a disabled/inactive button state being added to napari! You get so many hoorays from me!

While it surprised me a bit, I think I agree that the lower contrast/lower opacity option might be best here. I came to this conclusion because
- In terms of accessibility guidelines, [inactive states are typically exempt from minimum contrast](https://www.w3.org/TR/WCAG22/#non-text-contrast). While it’s nice to have it be legible, that means this isn’t a guide I need to push for this time.
- While it’s something I’d like to improve one day, the fact is napari has a lot of low contrast areas even when they are active (not limited to buttons). I do think it is possible to confuse the higher opacity option for another active state when there is no active state nearby to compare to.
- Even though it doesn’t quite match most of the buttons I’m seeing, I really like the button border for the inactive state. It’s a good way to indicate the area is a button. This makes me more comfortable with the low contrast having enough visual clarity.

Really, both options are similar to me so if there is another reason tipping the scale in favor of the higher contrast/higher opacity option I wouldn’t be opposed.

**Review by isabela-pf — APPROVED** [review]

**Review by LCObus — APPROVED** [review]
+1 to everything @isabela-pf said, and thank you for tagging us both! As we move forward I know she and I will continue to think big picture about ways to support viewer accessibility and design, but I love these fixes to ensure functional UI!

**Comment by andy-sweet** [comment]
@Czaki : given the new positive reviews, does this look good to you now? If so, feel free to merge when ready.

**Pull request merged**